ddp-workspace/projects/ddp-sdk/src/lib/components/activityForm/answers/activity-equation-answer/activityEquationAnswer.component.ts

```diff
@@ -0,0 +1,57 @@
+import { Component, Input, OnDestroy, OnInit } from '@angular/core';
+import { Subscription } from 'rxjs';
+import { map } from 'rxjs/operators';
+
+import { DecimalHelper } from '../../../../utility/decimalHelper';
+import { SubmissionManager } from '../../../../services/serviceAgents/submissionManager.service';
+import { DecimalAnswer } from '../../../../models/activity/decimalAnswer';
+import { ActivityEquationQuestionBlock } from '../../../../models/activity/activityEquationQuestionBlock';
+import { AnswerResponseEquation } from '../../../../models/activity/answerResponseEquation';
+
+@Component({
+  selector: 'ddp-activity-equation-answer',
+  template: `
+      <ddp-question-prompt [block]="block"></ddp-question-prompt>
+      <div class="equation-value">{{displayValue}}</div>
+  `
+})
+export class ActivityEquationAnswerComponent implements OnInit, OnDestroy {
+    @Input() block: ActivityEquationQuestionBlock;
+    private subscription: Subscription;
+
+    constructor(private submissionManager: SubmissionManager) {
+    }
+
+    ngOnInit(): void {
+        this.subscription = this.submissionManager.answerSubmissionResponse$.pipe(
+            map(response =>
+                (response.equations || []).filter(equation => this.block.stableId === equation.stableId)[0]
+            )
+        ).subscribe((equationToUpdate: AnswerResponseEquation) => {
+            equationToUpdate && this.block.setAnswer([equationToUpdate.values[this.block.compositeRowIndex]], false);
+        });
+    }
+
+    ngOnDestroy(): void {
+        this.subscription.unsubscribe();
+    }
+
+    get displayValue(): string {
+        return this.block.answer && this.block.answer[0] ? this.formatValue(this.block.answer[0]) : null;
+    }
+
+    private formatValue(answer: DecimalAnswer): string {
+        // align the answer decimal part
+        const fullValue = `${DecimalHelper.mapDecimalAnswerToNumber(answer)}`;
+        const [integerPart, decimalPart] = fullValue.split('.');
+        return integerPart + this.getFormattedDecimalPart(decimalPart);
+    }
```

Comment:
We have similar conversion code elsewhere, e.g.: https://github.com/broadinstitute/ddp-angular/blob/eff7a60134e39a99462d239cb648bb8650d58de1/ddp-workspace/projects/ddp-sdk/src/lib/components/activityForm/answers/activityNumericAnswer.component.ts#L123-L137
Let's consolidate into a utility. Or better: have DecimalAnswer interface have an implementation of toString().

Reply:
Actually, formatting of a `equation question answer` value and formatting of a `decimal question answer` differ a bit 
(even though they have the same type `DecimalAnswer`). 
In case of we don't have decimal part scale:
 - equation should keep the whole decimal part of the answer value (1.23 -> 1.23)
 - decimal should cut the whole decimal part (1.23 -> 1)
 
 Moreover, for `decimal question answer` when the component input fires a value update, we should transform just a number, e.g. 5.27 ( not `DecimalAnswer` format).
 
 Anyway, I have raised a separate [PR#1260](https://github.com/broadinstitute/ddp-angular/pull/1260) for that improvement. It is more generic one. However it did not become easier, imo. 
 
 What  are your thoughts?

Reply:
Merged that PR.